src/content/docs/en/guides/view-transitions.mdx

```diff
@@ -488,20 +576,24 @@ An event that fires immediately after the new page replaces the old page. You ca
 
 This event, when listened to on the **outgoing page**, is useful to pass along and restore any state on the DOM that needs to transfer over to the new page.
 
-For example, if you are implementing dark mode support, this event can be used to restore state across page loads:
+This is the latest point in the lifecyle where it is still safe to, for example, add a dark mode class name (`<html class="dark-mode">`), though you may wish to do so in an earlier event.
 
-```astro
+The "astro:after-swap" event occurs immediately after the browser history has been updated and the scroll position has been set. 
+
```

Comment:
```suggestion
Therefore, it is save now to override the standard scroll restoration if desired. The following example resets the horizontal and vertical scroll position to the top left corner of the page. 
```

Reply:
```astro
document.addEventListener('astro:after-swap', 
  () => window.scrollTo({ left: 0, top: 0, behavior: 'instant' }))
```

Reply:
Perfect, thanks, @martrapp  I'll add these in!

Just because GitHub is unintuitive and you might not realize you can add the code directly in your suggestion, here's my trick:

When you click to make a "suggestion", it will automatically format with three backticks. If you **increase the number of backticks to open and close your suggestions** (I add three, to make it six, out of habit), then you can have a regular three-backtick code sample directly in your suggestion! Like this!

![image](https://github.com/withastro/docs/assets/5098874/f0783b15-02ed-4109-80c8-e6aeade4786d)